src/scene/mesh-instance.js

```diff
@@ -455,6 +461,32 @@ class MeshInstance {
         return bindGroup;
     }
 
+    /**
+     * Performs a raycast with a given ray against the mesh
+     *
+     * @param {Ray} ray - the ray to perform the intersect test with
+     * @returns {Vec3} Null if there is no intersection or the point of intersection
+     * in world space
+     */
+    rayCast(ray) {
+        _worldTransformInverted.copy(this.node.getWorldTransform());
```

Comment:
avoid calling this twice. Do
`const worldTransform = this.node.getWorldTransform())';
and use it at the end of the function too